env0/resource_approval_policy.go

```diff
@@ -0,0 +1,149 @@
+package env0
+
+import (
+	"context"
+	"fmt"
+	"log"
+
+	"github.com/env0/terraform-provider-env0/client"
+	"github.com/google/uuid"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+const APPROVAL_POLICY = "approval-policy"
+
+func resourceApprovalPolicy() *schema.Resource {
+	return &schema.Resource{
+		CreateContext: resourceApprovalPolicyCreate,
+		ReadContext:   resourceApprovalPolicyRead,
+		UpdateContext: resourceApprovalPolicyUpdate,
+		DeleteContext: resourceApprovalPolicyDelete,
+
+		Importer: &schema.ResourceImporter{StateContext: resourceApprovalPolicyImport},
+
+		Schema: getTemplate(ApprovalPolicy),
+	}
+}
+
+func resourceApprovalPolicyCreate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	apiClient := meta.(client.ApiClientInterface)
+
+	var payload client.TemplateCreatePayload
+	if err := readResourceData(&payload, d); err != nil {
+		return diag.Errorf("schema resource data deserialization failed: %v", err)
+	}
+
+	payload.Type = APPROVAL_POLICY
+
+	template, err := apiClient.TemplateCreate(payload)
+	if err != nil {
+		return diag.Errorf("could not create approval policy template: %v", err)
+	}
+
+	d.SetId(template.Id)
+
+	return nil
+}
+
+func resourceApprovalPolicyRead(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	apiClient := meta.(client.ApiClientInterface)
+
+	approvalPolicy, err := apiClient.Template(d.Id())
+	if err != nil {
+		return ResourceGetFailure("approval policy", d, err)
+	}
+
+	if approvalPolicy.IsDeleted && !d.IsNewResource() {
```

Comment:
I'm not sure I's deleted is a property in Template. is it something that is added by the provider / TF?

Reply:
I copied it from the Template resource (just to be on the safe side). Do you think it's not required?

Reply:
oh I see. just checked out code, its logic that I wasn't aware of... lets keep is for safe measures 👍🏼